src/Microsoft.ServiceFabric.Services.Remoting/Client/ServiceProxyFactory.cs

```diff
@@ -91,65 +91,45 @@ public ServiceProxyFactory(OperationRetrySettings retrySettings = null)
         {
             var serviceInterfaceType = typeof(TServiceInterface);
 
-#if !DotNetCoreClr
-
             //Use provider to find the stack
-            if (this.proxyFactoryV1 == null && this.proxyFactoryV2 == null)
+            var provider = this.GetProviderAttribute(serviceInterfaceType);
```

Comment:
We get provider attribute each time service proxy factory is called. This is not a hot path, is it? So I don't think it matters for perf here?

Reply:
This is called when creating the proxy.

Reply:
Right! Which is fine, as long as this factory is only resued for services which use the same stacks (or have stacks that are compatible with each other).

However, ServiceProxy uses a [single instance](https://github.com/Mardoxx/service-fabric-services-and-actors-dotnet/blob/a66d4accddb40d214956bdb67db4ca9300fc9e62/src/Microsoft.ServiceFabric.Services.Remoting/Client/ServiceProxy.cs#L29).

The trouble with this is, if I do the following

```csharp
IV1Svc svc1 = ServiceProxy.Create<IV1Svc>(/*...*/);

// Internal DefaultServiceProxy is initialised with `proxyFactoryV1 != null`

IV2Svc svc2 = ServiceProxy.Create<IV2Svc>(/*...*/);

// When calling svc2.someMethod, it will not know to use v2 stack.
```

Do you see where I'm coming from here?

Reply:
yes, i see your point